lib/install.js

```diff
@@ -84,6 +86,24 @@ exports.install = function(targets, options) {
   options = options || {};
 
   return config.load()
+  .then(function () {
+    var hooks = config.pjson.hooks;
+    if (hooks) {
+      return new Loader().import(hooks.preinstall)
+        .then(function (m) {
+          if (m.default && typeof m.default === 'function') {
+            ui.log('info', 'Running preinstall hook...');
+            return Promise.resolve(m.default());
+          }
+        })
+        .catch(function (error) {
+          if (error) {
```

Comment:
Even if the error is undefined we'd still want the message to display. So perhaps just:

```javascript
if (error)
  ui.log('err', error);
```